### PR TITLE
MNT-23047: exifTool command defined as properties and dependant of ru…

### PR DESCRIPTION
**Pull request by vrmoreira**
…nning OS

### CODE REVIEW

**Comment by vrmoreira** [comment]
Some tests are still failing, therefore converting PR to a draft one

**Comment by vrmoreira** [comment]
Testes are green but have a few questions on this:
1. there are no Windows tests for ATS (it was tested manually)
2. When running IT against a Windows machine, they fail. IMO, it's weird

File: alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java
Comment (jamalkm) [inline alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java:100]: I don't see this bean is used anywhere.

File: pom.xml
Comment (jamalkm) [inline pom.xml:253]: Why we need this dependency?

**Review by jamalkm — COMMENTED** [review]
I'm not very familiar with ATS code, but the changes look OK to me, however I left a couple of comments.

File: pom.xml
Comment (vrmoreira) [reply pom.xml:253]: During the development of this PR, Veracode triggered a problem with that dependency, which didn't allowed to proceed the Travis build.

Meanwhile, I've implemented the same change on master, using [ACS-3492](https://alfresco.atlassian.net/browse/ACS-3492).

File: alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java
Comment (vrmoreira) [reply alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java]: "exifTool" removed. Thanks for pointing that out!

File: pom.xml
Comment (jamalkm) [reply pom.xml:253]: Where we get this dependency? If we already have snakeyaml in the transform-core then we should just upgrade it.

**Review by jamalkm — APPROVED** [review]
I'm on PTO until Sep 19th, so will not block the merge.

**Comment by vrmoreira** [comment]
@alandavis ,it is possible you to review this PR, to take a little break on the HxP fork? You've got an keep eye to spot defects :-)

**Comment by vrmoreira** [comment]
@alandavis , as requested (on Slack), I've fixed the following situations:
* `snakeyaml` was duplicated with the merge (fixed)
* as you said, that clone of the `InputStream` was weird and it was right: `parse` doesn't consume the inputstream, therefore makes no sense to clone it. Fixed.

Can you have another go on the review? Thanks!

File: engines/tika/src/main/java/org/alfresco/transform/tika/metadata/extractors/IPTCMetadataExtractor.java
Comment (alandavis) [inline engines/tika/src/main/java/org/alfresco/transform/tika/metadata/extractors/IPTCMetadataExtractor.java:60]: Are you sure this constructor is being called when SpringBoot is creating @Components?

File: engines/tika/src/main/java/org/alfresco/transform/tika/parsers/ExifToolParser.java
Comment (alandavis) [inline engines/tika/src/main/java/org/alfresco/transform/tika/parsers/ExifToolParser.java:99]: Please replace this comment with a blank line. It add nothing to the understanding and I was left wondering what the 'last chance' was.

File: engines/tika/src/main/java/org/alfresco/transform/tika/parsers/ExifToolParser.java
Comment (alandavis) [inline engines/tika/src/main/java/org/alfresco/transform/tika/parsers/ExifToolParser.java:91]: Please keep to the style of the class. We already have a mix in this project and adding: "if(" rather than: "if (" adds another. There is a second example of this on line 100.

**Review by alandavis — CHANGES_REQUESTED** [review]
See comments and fix minor changes.

Are you sure the new constructor is automatically being called? There are no tests which is a bit of a worry.

Good to see you found the RuntimeExec class for different OS command lines.

File: engines/tika/src/main/java/org/alfresco/transform/tika/metadata/extractors/IPTCMetadataExtractor.java
Comment (vrmoreira) [reply engines/tika/src/main/java/org/alfresco/transform/tika/metadata/extractors/IPTCMetadataExtractor.java:60]: Removed the default constructor.

File: engines/tika/src/main/java/org/alfresco/transform/tika/parsers/ExifToolParser.java
Comment (vrmoreira) [reply engines/tika/src/main/java/org/alfresco/transform/tika/parsers/ExifToolParser.java:91]: Think I didn't miss anything :)

File: engines/tika/src/main/java/org/alfresco/transform/tika/metadata/extractors/IPTCMetadataExtractor.java
Comment (vrmoreira) [reply engines/tika/src/main/java/org/alfresco/transform/tika/metadata/extractors/IPTCMetadataExtractor.java:61]: The `IPTCMetadataExtractorTest` uses this constructor and since it's the only constructor, think we've covered.
BTW, the parameter can be null (`ExifToolParser` will deal with nulls).

Comment (alandavis) [reply engines/tika/src/main/java/org/alfresco/transform/tika/metadata/extractors/IPTCMetadataExtractor.java:61]: Now the other constructor has been removed and the test class changed to use this one, I think we are fine

**Review by alandavis — APPROVED** [review]
LGTM

**Pull request merged**
